scalgrafanalib/core.py

```diff
@@ -99,19 +131,96 @@ def to_json_data(self):
+    text: str = attr.ib()
+    color: Optional[str] = attr.ib(default=None)
+    index: int = attr.ib(default=0)
+
+    def to_json_data(self) -> Dict[str, Any]:
+        result = {
+            "index": self.index,
+            "text": self.text,
+        }
+        if self.color:
+            result["color"] = self.color
+        return result
+
+
 @attr.s
 class StateTimeline(core.StateTimeline):
-    """StateTimeline: Allow settings minValue and maxValue"""
+    """StateTimeline: Allow settings minValue, maxValue, and color mappings for MongoDB states"""
 
     minValue = attr.ib(default=None)  # pylint: disable=invalid-name
     maxValue = attr.ib(default=None)  # pylint: disable=invalid-name
+    colorMode: str = attr.ib(default="palette-classic")  # pylint: disable=invalid-name
+    mappings: List[Union[StateMapping, Dict[str, Any]]] = attr.ib(
+        default=None
+    )  # pylint: disable=invalid-name
 
     def to_json_data(self) -> Json:
         json = super().to_json_data()
         if self.minValue:
             json["fieldConfig"]["defaults"]["min"] = self.minValue
         if self.maxValue:
             json["fieldConfig"]["defaults"]["max"] = self.maxValue
+
+        # Set color mode
+        if "fieldConfig" not in json:
+            json["fieldConfig"] = {}
+        if "defaults" not in json["fieldConfig"]:
+            json["fieldConfig"]["defaults"] = {}
+        if "color" not in json["fieldConfig"]["defaults"]:
+            json["fieldConfig"]["defaults"]["color"] = {}
+        json["fieldConfig"]["defaults"]["color"]["mode"] = self.colorMode
+
+        # Add mappings if provided
+        if self.mappings:
+            if "mappings" not in json["fieldConfig"]["defaults"]:
+                json["fieldConfig"]["defaults"]["mappings"] = []
+
+            # Create value mapping options
+            value_options = {}
+            for mapping in self.mappings:
+                # Handle both StateMapping objects and dictionaries
+                if isinstance(mapping, StateMapping):
+                    # StateMapping object
+                    key = str(mapping.value) if mapping.value is not None else "null"
+                    value_options[key] = mapping.to_json_data()
+                else:
+                    # Dictionary - convert to StateMapping format
+                    value = mapping.get("value")
+                    key = str(value) if value is not None else "null"
+                    result = {
+                        "index": mapping.get("index", 0),
+                        "text": mapping.get("text", ""),
+                    }
+                    if mapping.get("color"):
+                        result["color"] = mapping["color"]
+                    value_options[key] = result
+
+            # Add the value mapping
+            json["fieldConfig"]["defaults"]["mappings"].append(
+                {"type": "value", "options": value_options}
+            )
+
+            # Add range mapping for N/A values
```

Comment:
should not be a default, but specified explicitly where used ("N/A" may not be appropriate for all use cases)